### ISSUE TITLE
Capture rpm post script failures

### ISSUE THREAD

**Issue by schaefi** [SUSE/kiwi#539]
## Problem description

Vladimir Nadvornik reported:

> JFYI, there is a recent change in zypper which now reports exit code 107
> if post script fails. Before it ignored such errors.
> Maybe Kiwi should ignore this exit code to get the old behavior.

I think it would make sense to have a possibility in kiwi by a commandline switch to
ignore rpm post script errors.

**Comment by schaefi** [comment]
Potential issue with this idea:

* All the old versions of zypper out there do not distinguish the exit code by category and thus the switch in kiwi would sometimes work and sometimes not depending on which zypper exists on the host and in the image

**Comment by davidcassany** [comment]
> All the old versions of zypper out there do not distinguish the exit code by category and thus the switch in kiwi would sometimes work and sometimes not depending on which zypper exists on the host and in the image

I am not sure this is an issue we should really care, I mean having a different error handling depending on boostrap requests or system requests doesn't really look that bad to my eyes. As would expect to use this feature only as a temporary workaround in case there are packages that produce this kind of failures (imho this only reveals that there is something wrong with the package installation scripts). 

Also we could produce a warning if the exit code is > 100 and fail if is different to zero and < 100 when the mentioned flag is active, in case the mentioned flag is not active any != 0 would fail as it is now.

Probably it could also be a runtime config flag.

**Comment by Phazeus** [comment]
warning: %post(vkaudiosaver-2.0.6-1.x86_64) scriptlet failed, exit status 127
warning: %post(gnokii-0.6.31-13.10.x86_64) scriptlet failed, exit status 1

And KIWI just breaking. But this is not errors, just warnings and possible to continue... I has excluding packages:
vkaudiosaver xgnokii gnokii-lang gnokii gnome-bluetooth-plugin-phone-manager gnome-phone-manager gnome-phone-manager-lang
to be able to complete installation stage and then jut installing them with chrooting into image folder.

**Comment by Phazeus** [comment]
I suggest adding in global KIWI parameters special flag (e.g.  --ignore-zypper-errors) to force build process...

**Comment by Phazeus** [comment]
> Ah I think we should not do that because it would allow to create completely broken images, as there would be no way to distinguish a real error and the "others"

Yes, but this flag needs only when I got errors and check log and sure what this errors just warnings and possible to continue. And only in this case I can use such flag...  Just lazy installing this packages through chroot every times... :)

**Comment by davidcassany** [comment]
@scheafi yes in fact i am trying to find a way to restrict the change to some specific error code. At first I thought that the change was applying to any error code equal or higher than 100, then I realized I skimmed to fast the zypper docs, since we are only interested/worried about 107 (initially I thought the warnings were converted to 100 and up error codes, but its not the case).

The main problem we have here to code that in a generic way, as the error code is validated CommandProcess class not in any packager class, obviously we can't code the 107 match inside CommandProcess, now I having a look to the code to see if it looks reasonable to refactor Command and CommandProcess in a way that when the command is created we also inform about the error codes to ignore, or an error code filter function or something similar; in this way we could create Command.calls in kiwi/package_manager/zypper class with the information about which errorcode should be ignored. I am still on my way to see a clean feasible way to handle that, it not as simple as I initially thought.
